src/outbound.rs

```diff
@@ -153,8 +153,8 @@ impl Outbound {
 
     pub(crate) fn flush_loop(&self) {
         inject_delay();
-        let mut writer = self.writer.lock();
         loop {
+            let mut writer = self.writer.lock();
             while writer.flusher_should_wait() {
                 self.updated.wait(&mut writer);
```

Comment:
Do we drop lock inside of here?

Reply:
no, because this is a parking_lot mutex and condvar rather than a std one that takes the owned one. the compiler guarantees that we hold the mutex while accessing the writer held inside later on in the loop

Reply:
Ah yes, I think I had that in original just did not recognize..